packages/plugin-flex/src/commands/flex/plugins/deploy.ts

```diff
@@ -128,6 +137,37 @@ export default class FlexPluginsDeploy extends FlexPlugin {
     this.prints.deploySuccessful(this.pkg.name, pluginVersion.private ? 'private' : 'public', deployedData);
   }
 
+  /**
+   * Checks if there is already an uploaded asset with the same version and prompts user with an option to override if so
+   * @returns {Promise<boolean>}
+   */
+  async hasCollisionAndOverwrite(): Promise<boolean> {
+    const credentials = await getCredential();
+    const runtime = await getRuntime(credentials);
+    if (!runtime.environment) {
+      throw new FlexPluginError('No Runtime environment was found');
+    }
+
+    const pluginBaseUrl = getPaths().assetBaseUrlTemplate.replace('%PLUGIN_VERSION%', this.nextVersion as string);
+    const collision = runtime.build ? !_verifyPath(pluginBaseUrl, runtime.build) : false;
+
+    if (collision && !env.isCI()) {
```

Comment:
Move `env.isCI()` to the very top - no point fetching runtime/etc if this is CI. Then once this is moved, reverse the logic here too

```js
if (env.isCI()) {
  return false;
}

...
if (!collision) {
  return false;
}
const answer = await ...
```

Reply:
yep, was gonna add tests tmr! and good call, will update

Reply:
i'll update it, but why reverse the collision logic? do you just prefer having the simpler logic first?